docs/USAGE-GUIDE.md

```diff
@@ -843,15 +843,51 @@ that will be called after ClientHello was parsed.
 typedef int s2n_client_hello_fn(struct s2n_connection *conn, void *ctx);
 ```
 
-The callback function take as an input s2n-tls connection, which received
-ClientHello and context provided in **s2n_config_set_client_hello_cb**. The
-callback can get any ClientHello information from the connection and use
+The callback function takes a s2n-tls connection as input, which receives the
+ClientHello and the context previously provided in **s2n_config_set_client_hello_cb**.
+The callback can access any ClientHello information from the connection and use
 **s2n_connection_set_config** call to change the config of the connection.
```

Comment:
```suggestion
The callback can access any ClientHello information from the connection and use
the **s2n_connection_set_config** call to change the config of the connection.
```

Reply:
don't think `the` is needed here unless it was something like: `and use the **s2n_connection_set_config** method`

Reply:
How does `method` vs `call` change things?

Reply:
my bad, didn't notice the `call` suffix :D updated